src/exercise/06.js

```diff
@@ -2,30 +2,61 @@
 // http://localhost:3000/isolated/exercise/06.js
 
 import * as React from 'react'
-// 🐨 you'll want the following additional things from '../pokemon':
-// fetchPokemon: the function we call to get the pokemon info
-// PokemonInfoFallback: the thing we show while we're loading the pokemon info
-// PokemonDataView: the stuff we use to display the pokemon info
-import {PokemonForm} from '../pokemon'
+import {fetchPokemon, PokemonInfoFallback, PokemonDataView, PokemonForm} from '../pokemon'
 
 function PokemonInfo({pokemonName}) {
-  // 🐨 Have state for the pokemon (null)
-  // 🐨 use React.useEffect where the callback should be called whenever the
-  // pokemon name changes.
-  // 💰 DON'T FORGET THE DEPENDENCIES ARRAY!
-  // 💰 if the pokemonName is falsy (an empty string) then don't bother making the request (exit early).
-  // 🐨 before calling `fetchPokemon`, clear the current pokemon state by setting it to null
-  // 💰 Use the `fetchPokemon` function to fetch a pokemon by its name:
-  //   fetchPokemon('Pikachu').then(
-  //     pokemonData => {/* update all the state here */},
-  //   )
-  // 🐨 return the following things based on the `pokemon` state and `pokemonName` prop:
-  //   1. no pokemonName: 'Submit a pokemon'
-  //   2. pokemonName but no pokemon: <PokemonInfoFallback name={pokemonName} />
-  //   3. pokemon: <PokemonDataView pokemon={pokemon} />
-
-  // 💣 remove this
-  return 'TODO'
+  // Set up State Hooks.
+  // Tracks status of the application
+  const [status, setStatus] = React.useState('idle')
+  // Tracks the pokemon to be shown
+  const [pokemon, setPokemon] = React.useState(null)
+  // Tracks the error(s) to be shown
+  const [error, setError] = React.useState(null)
+
+  // Set up an Effect Hook to fetch the pokemon data only when `pokemonName` changes.
+  React.useEffect(() => {
+    // If the pokemonName is falsy, we don't need to fetch anything.
+    if (!pokemonName) {
+      return
+    }
+
+    // Indicate we're starting a new request.
+    setStatus('pending')
+    // Reset the pokemon to null so we can appropriately render the fallback.
```

Comment:
Look at this again later: resetting `pokemon` and `error` actually may not be necessary here.